In [703]:
import pandas as pd
import plotly.express as px 

Athlete_events = pd.read_csv("Data/athlete_events.csv")
Noc_regions = pd.read_csv("Data/noc_regions.csv")

df_athlete = pd.DataFrame(Athlete_events)
df_regions = pd.DataFrame(Noc_regions)
df_athlete.head()

ID                      Name Sex   Age  Height  Weight            Team  \
0   1                 A Dijiang   M  24.0   180.0    80.0           China   
1   2                  A Lamusi   M  23.0   170.0    60.0           China   
2   3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN         Denmark   
3   4      Edgar Lindenau Aabye   M  34.0     NaN     NaN  Denmark/Sweden   
4   5  Christine Jacoba Aaftink   F  21.0   185.0    82.0     Netherlands   

   NOC        Games  Year  Season       City          Sport  \
0  CHN  1992 Summer  1992  Summer  Barcelona     Basketball   
1  CHN  2012 Summer  2012  Summer     London           Judo   
2  DEN  1920 Summer  1920  Summer  Antwerpen       Football   
3  DEN  1900 Summer  1900  Summer      Paris     Tug-Of-War   
4  NED  1988 Winter  1988  Winter    Calgary  Speed Skating   

                              Event Medal  
0       Basketball Men's Basketball   NaN  
1      Judo Men's Extra-Lightweight   NaN  
2           Football Men's Football   NaN  
3       Tug-Of-War Men's Tug-Of-War  Gold  
4  Speed Skating Women's 500 metres   NaN

In [704]:
# Four sports to study, basketball, football, ice hockey, and boxing
# Create a dataframe for the sports of interest

In [705]:
df_sports = df_athlete[df_athlete['Sport'].isin(['Basketball', 'Football', 'Ice Hockey', 'Boxing'])]
df_sports.reset_index(drop=True, inplace=True)
df_sports.head()

ID                      Name Sex   Age  Height  Weight     Team  NOC  \
0   1                 A Dijiang   M  24.0   180.0    80.0    China  CHN   
1   3       Gunnar Nielsen Aaby   M  24.0     NaN     NaN  Denmark  DEN   
2   9          Antti Sami Aalto   M  26.0   186.0    96.0  Finland  FIN   
3  16  Juhamatti Tapio Aaltonen   M  28.0   184.0    85.0  Finland  FIN   
4  37        Ann Kristin Aarnes   F  23.0   182.0    64.0   Norway  NOR   

         Games  Year  Season            City       Sport  \
0  1992 Summer  1992  Summer       Barcelona  Basketball   
1  1920 Summer  1920  Summer       Antwerpen    Football   
2  2002 Winter  2002  Winter  Salt Lake City  Ice Hockey   
3  2014 Winter  2014  Winter           Sochi  Ice Hockey   
4  1996 Summer  1996  Summer         Atlanta    Football   

                         Event   Medal  
0  Basketball Men's Basketball     NaN  
1      Football Men's Football     NaN  
2  Ice Hockey Men's Ice Hockey     NaN  
3  Ice Hockey Men's Ice Hockey  Bronze  
4    Football Women's Football  Bronze

In [706]:
display(df_sports)

ID                        Name Sex   Age  Height  Weight  \
0           1                   A Dijiang   M  24.0   180.0    80.0   
1           3         Gunnar Nielsen Aaby   M  24.0     NaN     NaN   
2           9            Antti Sami Aalto   M  26.0   186.0    96.0   
3          16    Juhamatti Tapio Aaltonen   M  28.0   184.0    85.0   
4          37          Ann Kristin Aarnes   F  23.0   182.0    64.0   
...       ...                         ...  ..   ...     ...     ...   
22839  135530              Philip Zwiener   M  23.0   202.0    98.0   
22840  135538             Jonathan Zwikel   M  22.0   183.0    84.0   
22841  135538             Jonathan Zwikel   M  26.0   183.0    84.0   
22842  135566  James Francis "Jim" Zylker   M  21.0   175.0    75.0   
22843  135568      Olga Igorevna Zyuzkova   F  33.0   171.0    69.0   

                Team  NOC        Games  Year  Season            City  \
0              China  CHN  1992 Summer  1992  Summer       Barcelona   
1            Denmark  DEN  1920 Summer  1920  Summer       Antwerpen   
2            Finland  FIN  2002 Winter  2002  Winter  Salt Lake City   
3            Finland  FIN  2014 Winter  2014  Winter           Sochi   
4             Norway  NOR  1996 Summer  1996  Summer         Atlanta   
...              ...  ...          ...   ...     ...             ...   
22839        Germany  GER  2008 Summer  2008  Summer         Beijing   
22840         France  FRA  1998 Winter  1998  Winter          Nagano   
22841         France  FRA  2002 Winter  2002  Winter  Salt Lake City   
22842  United States  USA  1972 Summer  1972  Summer          Munich   
22843        Belarus  BLR  2016 Summer  2016  Summer  Rio de Janeiro   

            Sport                          Event   Medal  
0      Basketball    Basketball Men's Basketball     NaN  
1        Football        Football Men's Football     NaN  
2      Ice Hockey    Ice Hockey Men's Ice Hockey     NaN  
3      Ice Hockey    Ice Hockey Men's Ice Hockey  Bronze  
4        Football      Football Women's Football  Bronze  
...           ...                            ...     ...  
22839  Basketball    Basketball Men's Basketball     NaN  
22840  Ice Hockey    Ice Hockey Men's Ice Hockey     NaN  
22841  Ice Hockey    Ice Hockey Men's Ice Hockey     NaN  
22842    Football        Football Men's Football     NaN  
22843  Basketball  Basketball Women's Basketball     NaN  

[22844 rows x 15 columns]

In [707]:
# check for a specific year, specific country, specific sport with gold medal
df_sports[(df_sports['Year'] == 2000) & (df_sports['NOC'] == 'USA') & (df_sports['Sport'] == 'Basketball') & (df_sports['Medal'] == 'Gold')]

ID                                  Name Sex   Age  Height  Weight  \
73        351            Julius Shareef Abdur-Rahim   M  23.0   202.0   104.0   
582      2863                      Walter Ray Allen   M  25.0   192.0    93.0   
1425     7040            Vincent Lamont "Vin" Baker   M  28.0   207.0   109.0   
2367    13042  Alice Ruth "Ruthie" Bolton-Holifield   F  33.0   173.0    68.0   
3272    18662          Vincent Lamar "Vince" Carter   M  23.0   195.0   102.0   
5305    31507                        Teresa Edwards   F  36.0   175.0    68.0   
6587    38826                 Kevin Maurice Garnett   M  24.0   207.0   100.0   
7207    42871               Yolanda Yvette Griffith   F  30.0   193.0    81.0   
7694    45964          Timothy Duane "Tim" Hardaway   M  34.0   180.0    88.0   
8373    50268                    Allan Wade Houston   M  29.0   195.0    91.0   
10010   59569                  Jason Frederick Kidd   M  27.0   193.0    96.0   
11501   68985       Lisa Deshawn Leslie (-Lockwood)   F  28.0   196.0    77.0   
13024   77407               Nikki Kensengane McCray   F  28.0   181.0    72.0   
13036   77496             Antonio Keithflen McDyess   M  26.0   202.0   109.0   
13451   80053          DeLisha Lachell Milton-Jones   F  26.0   185.0    84.0   
13916   82625               Alonzo Harding Mourning   M  30.0   206.0   117.0   
15753   92435                    Gary Dwayne Payton   M  32.0   190.0    86.0   
19158  112322           Katherine May "Katie" Smith   F  26.0   180.0    79.0   
19167  112404           Steven Delano "Steve" Smith   M  31.0   200.0   100.0   
19502  114217                  Dawn Michelle Staley   F  30.0   165.0    58.0   
19998  117168                 Sheryl Denice Swoopes   F  29.0   182.0    72.0   
22076  130646                 Natalie Jean Williams   F  29.0   188.0    80.0   
22156  131390      Kara Elizabeth Wolters (-Drinan)   F  25.0   201.0    95.0   

                Team  NOC        Games  Year  Season    City       Sport  \
73     United States  USA  2000 Summer  2000  Summer  Sydney  Basketball   
582    United States  USA  2000 Summer  2000  Summer  Sydney  Basketball   
1425   United States  USA  2000 Summer  2000  Summer  Sydney  Basketball   
2367   United States  USA  2000 Summer  2000  Summer  Sydney  Basketball   
3272   United States  USA  2000 Summer  2000  Summer  Sydney  Basketball   
5305   United States  USA  2000 Summer  2000  Summer  Sydney  Basketball   
6587   United States  USA  2000 Summer  2000  Summer  Sydney  Basketball   
7207   United States  USA  2000 Summer  2000  Summer  Sydney  Basketball   
7694   United States  USA  2000 Summer  2000  Summer  Sydney  Basketball   
8373   United States  USA  2000 Summer  2000  Summer  Sydney  Basketball   
10010  United States  USA  2000 Summer  2000  Summer  Sydney  Basketball   
11501  United States  USA  2000 Summer  2000  Summer  Sydney  Basketball   
13024  United States  USA  2000 Summer  2000  Summer  Sydney  Basketball   
13036  United States  USA  2000 Summer  2000  Summer  Sydney  Basketball   
13451  United States  USA  2000 Summer  2000  Summer  Sydney  Basketball   
13916  United States  USA  2000 Summer  2000  Summer  Sydney  Basketball   
15753  United States  USA  2000 Summer  2000  Summer  Sydney  Basketball   
19158  United States  USA  2000 Summer  2000  Summer  Sydney  Basketball   
19167  United States  USA  2000 Summer  2000  Summer  Sydney  Basketball   
19502  United States  USA  2000 Summer  2000  Summer  Sydney  Basketball   
19998  United States  USA  2000 Summer  2000  Summer  Sydney  Basketball   
22076  United States  USA  2000 Summer  2000  Summer  Sydney  Basketball   
22156  United States  USA  2000 Summer  2000  Summer  Sydney  Basketball   

                               Event Medal  
73       Basketball Men's Basketball  Gold  
582      Basketball Men's Basketball  Gold  
1425     Basketball Men's Basketball  Gold  
2367   Basketball Women's Basketball  Gold  
3272     Basketb

Note: the Unified Team is the name used for the sports team of the former Soviet Union at the 1992 Winter Olympics in Albertville and the 1992 Summer Olympics in Barcelona
- source: https://en.wikipedia.org/wiki/Unified_Team_at_the_Olympics

In [708]:
# check for missing values in Medal column
no_medals = df_sports['Medal'].isnull().sum()
print(f"There are {no_medals} missing values in the Medal column")

no_medals_percentage = no_medals/len(df_sports) * 100
print(f"The percentage of missing values in the Medal column is {no_medals_percentage:.2f}%")

There are 17719 missing values in the Medal column
The percentage of missing values in the Medal column is 77.57%


In [709]:
# check for missing values in Age column
no_age = df_sports["Age"].isnull().sum()
print(f"There are {no_age} missing values in the Age column")

no_age_percentage = no_age / len(df_sports) * 100
print(f"The percentage of missing values in the Age column is {no_age_percentage:.2f}%")

There are 908 missing values in the Age column
The percentage of missing values in the Age column is 3.97%


- As we can see, that the proportion of missing values in the Age column is about 3.97% of the total number of rows in the dataframe.
- So we can assume that is ok to replace the missing values with the mean of the Age column.

In [710]:
# replace the missing values in age column with the mean age
df_sports["Age"].fillna(df_sports["Age"].mean(), inplace=True)

# convert the Age column to int
df_sports["Age"] = df_sports["Age"].astype(int)


C:\Users\hadda\AppData\Local\Temp\ipykernel_17260\1433037379.py:2: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy

C:\Users\hadda\AppData\Local\Temp\ipykernel_17260\1433037379.py:5: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy



- After thorough examination, the missing values are not missing at random, they are missing because the athlete did not win a medal.

In [711]:
df_boxing = df_sports[df_sports["Sport"] == "Boxing"]
df_boxing.head()

ID                   Name Sex  Age  Height  Weight      Team  NOC  \
6   48       Abdelhak Aatakni   M   24     NaN    64.0   Morocco  MAR   
7   54  Mohamed Jamshid Abadi   M   24     NaN     NaN      Iran  IRI   
8   57     Borislav Abadzhiev   M   24   176.0    66.0  Bulgaria  BUL   
10  66        Mohamed Abakkar   M   24   156.0    48.0     Sudan  SUD   
13  77          Antoine Abang   M   27   166.0    75.0  Cameroon  CMR   

          Games  Year  Season         City   Sport  \
6   2012 Summer  2012  Summer       London  Boxing   
7   1948 Summer  1948  Summer       London  Boxing   
8   1988 Summer  1988  Summer        Seoul  Boxing   
10  1972 Summer  1972  Summer       Munich  Boxing   
13  1968 Summer  1968  Summer  Mexico City  Boxing   

                              Event Medal  
6   Boxing Men's Light-Welterweight   NaN  
7          Boxing Men's Heavyweight   NaN  
8   Boxing Men's Light-Welterweight   NaN  
10           Boxing Men's Flyweight   NaN  
13        Boxing Men's Middleweight   NaN

In [712]:
df_boxing["Season"].unique() # boxing only has Summer season

array(['Summer'], dtype=object)

In [713]:
# since we are trying to visualize the medal distribution for each land(not athlete) we have to:
# exclude the athletes who did not win a medal from the dataframe 

In [714]:
# remove all columns except Age, Team and Medal
df_boxing = df_boxing[["Age", "Team", "Medal"]]

# remove all rows where the medal is NaN
df_boxing = df_boxing[df_boxing["Medal"].notna()] # notna() is the same as isnull() == False
df_boxing

Age          Team   Medal
69      26    Uzbekistan    Gold
134     21   Puerto Rico  Bronze
140     29       Morocco  Bronze
142     27       Morocco  Bronze
149     23          Cuba  Bronze
...    ...           ...     ...
22788   27         China    Gold
22789   31         China    Gold
22813   20         Italy  Silver
22821   22  East Germany    Gold
22836   21       Belarus  Silver

[944 rows x 3 columns]

In [715]:
# use groupby to group the data by Team and Medal
df_boxing_medal = df_boxing.groupby(["Team", "Medal"]) 
df_boxing_medal = df_boxing_medal.size() # size() counts the number of rows in each group
df_boxing_medal = df_boxing_medal.reset_index(name="Count") # reset_index(name="Count") resets the index and adds a new column named Count
# sort by count of gold medals
df_boxing_medal = df_boxing_medal.sort_values(by=["Count"], ascending=False).reset_index(drop=True)
df_boxing_medal

Team   Medal  Count
0    United States    Gold     50
1    United States  Bronze     39
2             Cuba    Gold     37
3           Poland  Bronze     26
4    Great Britain  Bronze     25
..             ...     ...    ...
167         Guyana  Bronze      1
168          Ghana  Silver      1
169        Georgia  Bronze      1
170        Finland  Silver      1
171         Zambia  Bronze      1

[172 rows x 3 columns]

In [716]:
# plot the data change the color of medel to gold, silver, bronze.
fig = px.bar(
    df_boxing_medal,
    x="Team",
    y="Count",
    log_y=True,
    color="Medal",
    # barmode="group",
    category_orders={"Medal": ["Gold", "Silver", "Bronze"]},
    color_discrete_sequence=["gold", "silver", "brown"],
    title="Medal distribution for boxing",
)
fig.show()
# not all countries are shown on the x-axis

In [717]:
# plot all medals won by each country
fig = px.histogram(
    df_boxing,
    x="Team",
    log_y=True,
    barmode="group",
    color="Medal",
    title="Medals won by each country in boxing",
)
fig.show()

# as we notice that including all the countries in the plot makes it hard to read

In [718]:
# keep only gold medals
df_boxing_gold = df_boxing[df_boxing["Medal"] == "Gold"]
df_boxing_gold

Age           Team Medal
69      26     Uzbekistan  Gold
174     29  Great Britain  Gold
175     33  Great Britain  Gold
201     20  United States  Gold
486     24           Cuba  Gold
...    ...            ...   ...
22698   22     Kazakhstan  Gold
22777   23     Uzbekistan  Gold
22788   27          China  Gold
22789   31          China  Gold
22821   22   East Germany  Gold

[252 rows x 3 columns]

In [719]:
# plot the distribution of gold medals for each team
fig = px.histogram(
    df_boxing_gold,
    x="Team",
    log_y=True,
    color="Team",
    title="Distribution of gold medals for each team in boxing",
    labels={"Team": "Team", "count": "Number of gold medals"}, # why can't I change the count label? 
)
fig.show()

In [720]:
# keep only silver medals
df_boxing_silver = df_boxing[df_boxing["Medal"] == "Silver"]
df_boxing_silver

Age        Team   Medal
176     23      Poland  Silver
299     25      Sweden  Silver
485     20        Cuba  Silver
519     26     Romania  Silver
669     29       Egypt  Silver
...    ...         ...     ...
22716   23  Kazakhstan  Silver
22717   27  Kazakhstan  Silver
22745   26     Germany  Silver
22813   20       Italy  Silver
22836   21     Belarus  Silver

[251 rows x 3 columns]

In [721]:
# plot the distribution of silver medals for each team
fig = px.histogram(
    df_boxing_silver,
    x="Team",
    log_y=True,
    color="Team",
    title="Distribution of silver medals for each team in boxing",
    labels={"Team": "Team", "count": "Number of silver medals"},
)
fig.show()

In [722]:
# keep only bronze medals
df_boxing_bronze = df_boxing[df_boxing["Medal"] == "Bronze"]    
df_boxing_bronze

Age          Team   Medal
134     21   Puerto Rico  Bronze
140     29       Morocco  Bronze
142     27       Morocco  Bronze
149     23          Cuba  Bronze
164     23        Poland  Bronze
...    ...           ...     ...
22540   27        Mexico  Bronze
22543   29        Poland  Bronze
22739   23  West Germany  Bronze
22748   28       Romania  Bronze
22787   23         China  Bronze

[441 rows x 3 columns]

In [723]:
# plot the distribution of bronze medals for each team
fig = px.histogram(
    df_boxing_bronze,
    x="Team",
    log_y=True,
    color="Team",
    title="Distribution of bronze medals for each team in boxing",
    labels={"Team": "Team", "count": "Number of bronze medals"},
)
fig.show()

In [724]:
# plot the age distribution in teams
# i went with scatter plot because it is easier to read than a histogram
fig = px.scatter(
    df_boxing,
    x="Team",
    y="Age",
    color="Team",
    title="Age distribution in boxing teams",
)
fig.show()

In [725]:
# plot the age distribution in teams using box plot
fig = px.box(
    df_boxing,
    x="Team",
    y="Age",
    color="Team",
    title="Age distribution in boxing teams",
)
fig.show()

In [726]:
# violin plot
fig = px.violin(
    df_boxing,
    x="Team",
    y="Age",
    color="Team",
    title="Age distribution in boxing teams",
)
fig.show()

In [727]:
# sort by land that won most medals
# medel värde till gold medal 
# kön fördelning 

In [728]:
# # i main när en callback funktion körs för att ändra inställning på grafen:
# # (detta händer alltså varje gång någon klickar på en knapp som ändrar inställningar)

# # exempel på variabler för att ändra på grafen:
# x, y = "År", "Antal Deltagare"
# sortering, log = ("Summer", ""), False
# title = f"{y} per {x}"

# data = df[df["Årstid"].isin([*sortering])] # dataframen sorteras enligt någon variabel
# px.bar(data, x = x, y = y, log_y = log, title = title) # grafen plottas med specifika series för x och y (notera att vi också kan modifiera data här)

In [729]:
df_basketball = df_sports[df_sports["Sport"] == "Basketball"]
df_basketball.head()

ID                                Name Sex  Age  Height  Weight   Team  \
0     1                           A Dijiang   M   24   180.0    80.0  China   
11   69                   Tamara Abalde Daz   F   19   185.0    72.0  Spain   
23  124               Youssef Mohamed Abbas   M   31     NaN     NaN  Egypt   
25  136                    Alessandro Abbio   M   29   195.0    85.0  Italy   
33  192  Ahmed El-Sayed Abdel Hamid Mobarak   M   25   189.0    85.0  Egypt   

    NOC        Games  Year  Season       City       Sport  \
0   CHN  1992 Summer  1992  Summer  Barcelona  Basketball   
11  ESP  2008 Summer  2008  Summer    Beijing  Basketball   
23  EGY  1952 Summer  1952  Summer   Helsinki  Basketball   
25  ITA  2000 Summer  2000  Summer     Sydney  Basketball   
33  EGY  1972 Summer  1972  Summer     Munich  Basketball   

                            Event Medal  
0     Basketball Men's Basketball   NaN  
11  Basketball Women's Basketball   NaN  
23    Basketball Men's Basketball   NaN  
25    Basketball Men's Basketball   NaN  
33    Basketball Men's Basketball   NaN

In [730]:
# remove all nan rows in the Medal column
df_basketball = df_basketball[df_basketball["Medal"].notna()]
df_basketball

ID                             Name Sex  Age  Height  Weight  \
73        351       Julius Shareef Abdur-Rahim   M   23   202.0   104.0   
122       576  Alejandro "lex" Abrines Redondo   M   23   198.0    93.0   
124       580     Svetlana Olegovna Abrosimova   F   28   188.0    77.0   
152       727      Martn Ral Acosta y Lara Daz   M   27     NaN     NaN   
206      1000                Dainius Adomaitis   M   26   202.0   100.0   
...       ...                              ...  ..  ...     ...     ...   
22802  135345       Viktor Alekseyevich Zubkov   M   19   202.0    88.0   
22803  135345       Viktor Alekseyevich Zubkov   M   23   202.0    88.0   
22816  135401               Eurelijus ukauskas   M   22   218.0   115.0   
22817  135401               Eurelijus ukauskas   M   27   218.0   115.0   
22819  135402               Mindaugas ukauskas   M   20   201.0   100.0   

                Team  NOC        Games  Year  Season            City  \
73     United States  USA  2000 Summer  2000  Summer          Sydney   
122            Spain  ESP  2016 Summer  2016  Summer  Rio de Janeiro   
124           Russia  RUS  2008 Summer  2008  Summer         Beijing   
152          Uruguay  URU  1952 Summer  1952  Summer        Helsinki   
206        Lithuania  LTU  2000 Summer  2000  Summer          Sydney   
...              ...  ...          ...   ...     ...             ...   
22802   Soviet Union  URS  1956 Summer  1956  Summer       Melbourne   
22803   Soviet Union  URS  1960 Summer  1960  Summer            Roma   
22816      Lithuania  LTU  1996 Summer  1996  Summer         Atlanta   
22817      Lithuania  LTU  2000 Summer  2000  Summer          Sydney   
22819      Lithuania  LTU  1996 Summer  1996  Summer         Atlanta   

            Sport                          Event   Medal  
73     Basketball    Basketball Men's Basketball    Gold  
122    Basketball    Basketball Men's Basketball  Bronze  
124    Basketball  Basketball Women's Basketball  Bronze  
152    Basketball    Basketball Men's Basketball  Bronze  
206    Basketball    Basketball Men's Basketball  Bronze  
...           ...                            ...     ...  
22802  Basketball    Basketball Men's Basketball  Silver  
22803  Basketball    Basketball Men's Basketball  Silver  
22816  Basketball    Basketball Men's Basketball  Bronze  
22817  Basketball    Basketball Men's Basketball  Bronze  
22819  Basketball    Basketball Men's Basketball  Bronze  

[1080 rows x 15 columns]

In [731]:
# how many nan values in height and weight?
height_basketball = df_basketball["Height"].isnull().sum() 
weight_basketball = df_basketball["Weight"].isnull().sum()
no_height_percentage = round(height_basketball / len(df_basketball) * 100, 2)
no_weight_percentage = round(weight_basketball / len(df_basketball) * 100, 2)
print(f"The percentage of athletes with no height is {no_height_percentage}%, and the percentage of athletes with no weight is {no_weight_percentage}%.")

The percentage of athletes with no height is 5.19%, and the percentage of athletes with no weight is 7.41%.


In [750]:
# replace all nan values in height and weight with the mean
df_basketball["Height"].fillna(df_basketball["Height"].mean(), inplace=True)
df_basketball["Weight"].fillna(df_basketball["Weight"].mean(), inplace=True)

# convert height and weight to int
df_basketball["Height"] = df_basketball["Height"].astype(int)
df_basketball["Weight"] = df_basketball["Weight"].astype(int)

ID                             Name Sex  Age  Height  Weight  \
326      1530          Gordon "Gord" Aitchison   M   27     191      86   
589      2909        Ian Alistair "Al" Allison   M   27     191      86   
3466    19836   Arthur St. Clair "Art" Chapman   M   23     191      86   
3469    19838  Charles Winston "Chuck" Chapman   M   25     188      86   
4440    25790          Edward John "Ed" Dawson   M   28     191      86   
13244   78873          Irving "Toots" Meretsky   M   24     191      86   
15774   92595       James Douglas "Doug" Peden   M   20     183      86   
19640  115153            James "Jimmy" Stewart   M   26     191      86   
22122  131117     Malcolm Edward "Red" Wiseman   M   23     191      86   

         Team  NOC        Games  Year  Season    City       Sport  \
326    Canada  CAN  1936 Summer  1936  Summer  Berlin  Basketball   
589    Canada  CAN  1936 Summer  1936  Summer  Berlin  Basketball   
3466   Canada  CAN  1936 Summer  1936  Summer  Berlin  Basketball   
3469   Canada  CAN  1936 Summer  1936  Summer  Berlin  Basketball   
4440   Canada  CAN  1936 Summer  1936  Summer  Berlin  Basketball   
13244  Canada  CAN  1936 Summer  1936  Summer  Berlin  Basketball   
15774  Canada  CAN  1936 Summer  1936  Summer  Berlin  Basketball   
19640  Canada  CAN  1936 Summer  1936  Summer  Berlin  Basketball   
22122  Canada  CAN  1936 Summer  1936  Summer  Berlin  Basketball   

                             Event   Medal  
326    Basketball Men's Basketball  Silver  
589    Basketball Men's Basketball  Silver  
3466   Basketball Men's Basketball  Silver  
3469   Basketball Men's Basketball  Silver  
4440   Basketball Men's Basketball  Silver  
13244  Basketball Men's Basketball  Silver  
15774  Basketball Men's Basketball  Silver  
19640  Basketball Men's Basketball  Silver  
22122  Basketball Men's Basketball  Silver

In [740]:
# plot medal distribution for each team
fig = px.histogram(
    df_basketball,
    x="Team",
    log_y=True,
    barmode="group",
    color="Medal",
    title="Medals won by each country in basketball",
)
fig.show()

In [754]:
# we are getting 281 gold medels for the US team, while the men won 16 gold medals and the women won 9
# so the total should be 25

# canada had 9 players
df_basketball[df_basketball["Team"] == "Canada"]

# remove all columns, except Team, Games, Year, Season, City, Event, Medal
df_basketball = df_basketball[["Team", "Games", "Year", "Season", "City", "Event", "Medal"]]
df_basketball

# join similar teams, with similar games, year, season, city, event and medal type
df_basketball = df_basketball.groupby(["Team", "Games", "Year", "Season", "City", "Event", "Medal"]).size().reset_index(name="Count")
df_basketball.head(60)

Team        Games  Year  Season            City  \
0               Argentina  2004 Summer  2004  Summer          Athina   
1               Argentina  2008 Summer  2008  Summer         Beijing   
2               Australia  1996 Summer  1996  Summer         Atlanta   
3               Australia  2000 Summer  2000  Summer          Sydney   
4               Australia  2004 Summer  2004  Summer          Athina   
5               Australia  2008 Summer  2008  Summer         Beijing   
6               Australia  2012 Summer  2012  Summer          London   
7                  Brazil  1948 Summer  1948  Summer          London   
8                  Brazil  1960 Summer  1960  Summer            Roma   
9                  Brazil  1964 Summer  1964  Summer           Tokyo   
10                 Brazil  1996 Summer  1996  Summer         Atlanta   
11                 Brazil  2000 Summer  2000  Summer          Sydney   
12               Bulgaria  1976 Summer  1976  Summer        Montreal   
13               Bulgaria  1980 Summer  1980  Summer          Moskva   
14                 Canada  1936 Summer  1936  Summer          Berlin   
15                  China  1984 Summer  1984  Summer     Los Angeles   
16                  China  1992 Summer  1992  Summer       Barcelona   
17                Croatia  1992 Summer  1992  Summer       Barcelona   
18                   Cuba  1972 Summer  1972  Summer          Munich   
19                 France  1948 Summer  1948  Summer          London   
20                 France  2000 Summer  2000  Summer          Sydney   
21                 France  2012 Summer  2012  Summer          London   
22                  Italy  1980 Summer  1980  Summer          Moskva   
23                  Italy  2004 Summer  2004  Summer          Athina   
24              Lithuania  1992 Summer  1992  Summer       Barcelona   
25              Lithuania  1996 Summer  1996  Summer         Atlanta   
26              Lithuania  2000 Summer  2000  Summer          Sydney   
27                 Mexico  1936 Summer  1936  Summer          Berlin   
28                 Russia  2004 Summer  2004  Summer          Athina   
29                 Russia  2008 Summer  2008  Summer         Beijing   
30                 Russia  2012 Summer  2012  Summer          London   
31                 Serbia  2016 Summer  2016  Summer  Rio de Janeiro   
32                 Serbia  2016 Summer  2016  Summer  Rio de Janeiro   
33  Serbia and Montenegro  1996 Summer  1996  Summer         Atlanta   
34            South Korea  1984 Summer  1984  Summer     Los Angeles   
35           Soviet Union  1952 Summer  1952  Summer        Helsinki   
36           Soviet Union  1956 Summer  1956  Summer       Melbourne   
37           Soviet Union  1960 Summer  1960  Summer            Roma   
38           Soviet Union  1964 Summer  1964  Summer           Tokyo   
39           Soviet Union  1968 Summer  1968  Summer     Mexico City   
40           Soviet Union  1972 Summer  1972  Summer          Munich   
41           Soviet Union  1976 Summer  1976  Summer        Montreal   
42           Soviet Union  1976 Summer  1976  Summer        Montreal   
43           Soviet Union  1980 Summer  1980  Summer          Moskva   
44           Soviet Union  1980 Summer  1980  Summer          Moskva   
45           Soviet Union  1988 Summer  1988  Summer           Seoul   
46           Soviet Union  1988 Summer  1988  Summer           Seoul   
47                  Spain  1984 Summer  1984  Summer     Los Angeles   
48                  Spain  2008 Summer  2008  Summer         Beijing   
49                  Spain  2012 Summer  2012  Summer          London   
50                  Spain  2016 Summer  2016  Summer  Rio de Janeiro   
51                  Spain  2016 Summer  2016  Summer  Rio de Janeiro   
52           Unified Team  1992 Summer  1992  Summer       Barcelona   
53          United States  1936 Summer  1936  Summer          Berlin   
54          United States  1948 Summer  1948  Summer         